stdlib/src/builtin/_hasher.mojo

```diff
@@ -0,0 +1,40 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+
+trait _HashableWithHasher:
+    fn __hash__[H: _Hasher](self, inout hasher: H):
+        ...
+
+
+trait _Hasher:
```

Comment:
**Question** Are we knowingly omitting the `_update_with_bytes` API from https://github.com/modularml/mojo/blob/32e7c06bd832f9e781d00bf11cc6911c7b678c18/proposals/improved-hash-module.md#introduce-a-hasher-trait for now?

Reply:
Yeah we'll add it in a future PR, that is intentional

Reply:
SG, I'll land this now internally. Thanks!